### ISSUE TITLE
When the hypervisor extension is implemented, the following are also mandatory

### ISSUE THREAD

**Issue by mark-riscv** [riscv/riscv-profiles#93]
does this mean these are included in H? we simplified the spreadsheet to have 2 columns: included , implied. currently we have the following things marked mandatory in RV22S64 but it sounds like that is wrong:

Ssstateen Supervisor-mode view of the state-enable extension. The supervisor-mode (sstateen0-3) and hypervisor-mode (hstateen0-3) state-enable registers must be provided.

Shcounterenw For any hpmcounter that is not read-only zero, the corresponding bit in hcounteren must be writable.

Shvstvala vstval must be written in all cases described above for stval.

Shtvala htval must be written with the faulting guest physical address in all circumstances permitted by the ISA.

Shvstvecd vstvec.MODE must be capable of holding the value 0 (Direct). When vstvec.MODE=Direct, vstvec.BASE must be capable of holding any valid four-byte-aligned address.

Shvsatpa All translation modes supported in satp must be supported in vsatp.

Shgatpa For each supported virtual memory scheme SvNN supported in satp, the corresponding hgatp SvNNx4 mode must be supported. The hgatp mode Bare must also be supported

**Comment by a4lg** [comment]
TL;DR answer: no, they are not a part of `H`.

It's not "*extension* requires/implies/includes *extension*" thing.  It means "*profile* requires *extension*" (under some circumstances; i.e. `H` is implemented).

All `Sh*` extensions defined in the RISC-V Profiles denote certain property of the hypervisor extension implementation (which minimum `H` implementation are not required to do so).  In other words, those `Sh*` extensions are *additional* restrictions to the minimal `H` extension.  If we use the term "imply", `Sh*` extensions may imply `H` but not vice versa.

`Ssstateen` extension is a completely separate extension from `H` (even `Ssstateen` without `H` is possible in theory) but provides some flexibility running the guest operating system if `H` is implemented.  I'm not completely sure about the reason this is mandated when `H` is implemented but having it is helpful anyway.
